### DISCUSSION TITLE
Wrong graphic data

### DISCUSSION THREAD

**Discussion by R3Cardoso** [wlcrs/huawei_solar#377] [General]
On the energy dashboard I noticed that it says my house consume energy from solar power and batery during the night.
I got a 10khw battery that provided enough energy to power my house during the night, why is the dash board saying it consume power from the panels.
On Fusion solar app it shows correctly, showing I all the energy need was consumed came from the battery during the night.

**Comment by sLesage** [comment]
Could this be because the Inverter Daily Yield <> Daily production from your solar panels ? I've noticed that it's actualyy also including the power which has been drawn from the battery in the Inverter's Daily Yield

**Comment by R3Cardoso** [comment]
It improved, and looks correct now after paste the code  solar_energy_riemann on configuration.
Still got allways 0.1/0.2kwh from the power grid, but I assume that is normal.

**Comment by MG-Programmer** [comment]
Hello,

I have the same issue. Can you please explain it in more detail how you solved it? Especially with the solar_energy_riemann integration

Thanks in advance
Marco

**Reply by R3Cardoso** [reply]
Read this solution: https://github.com/wlcrs/huawei_solar/wiki/Daily-Solar-Yield

And past the two last codes on configuration.yaml.
It will start right way showing the right data, data before that will stay as before.

**Reply by MG-Programmer** [reply]
Thank you. The tip was helpful. Now the inverter shows the right values.

**Comment by MG-Programmer** [comment]
I have another question. I think I have configured the battery wrong in the energy dashboard. In my SolarFusion App it shows that the battery is full and so there will no charge or discharge current flow. In the energy dashboard it shows a permanent charge and discharge.
How did you config this correct?
![FusionSolar energy flow](https://github.com/wlcrs/huawei_solar/assets/27810898/6c7a7a36-e6c6-4304-aa0a-1549ec7724e0)
![HA energy flow](https://github.com/wlcrs/huawei_solar/assets/27810898/798e5cb1-8d87-45a3-afa1-1bc67131a9c6)
![Screenshot 2023-05-25 095105](https://github.com/wlcrs/huawei_solar/assets/27810898/729cc268-f4ae-43cf-8afd-246a08294448)

**Reply by R3Cardoso** [reply]
On HA mine is like yours shows allways a in/out flow.
On solar app ,mine never reach 100%.
On the configuration on HA is like yours.

As far as I read on that solution I share, the solar app and HA values will never be 100% equal.

**Reply by sLesage** [reply]
@MG-Programmer can you explain how you did that ? I've never been able to make everything work correctly yet. I did try seting up the Energy Dashboard, but can't seem to get that working correctly

![image](https://github.com/wlcrs/huawei_solar/assets/530745/907d62cd-af7c-4861-900b-36b5d6f6a403)

And I've tried  many things, even a custom dashboard

![image](https://github.com/wlcrs/huawei_solar/assets/530745/3d0c3aea-c3e4-4eb7-897f-157160514585)

Doesn't matter what I do, I havn't been able to figure it out quite right. Like it shows 776 w being produced now, Fusion Solar reports 881W. That's only a amall difference though. Battery shows as 100% in the graphic, while HA numbers show 76%. I am consuming 3378 W from the battery right now, so 3.4 is close. Additionally I'm pulling some power from the grid, but that is only 0.03 kW at this time.

Also experimenting with something else I have found here somewhere

![image](https://github.com/wlcrs/huawei_solar/assets/530745/ad345e1b-a890-49c6-a12b-f36728f061eb)

Numbers in there are closer in the Power Flows diagram, but not 100% accurate. I know I still need to fiddle with the Solar Yield things, but my Solar Yield is pretty close there 29.93 in HA 29.86 in Fusion Solar, 5.35 Exported today in HA 5.43 in Fusion Solar. House consumption for today in HA 17.54 in Fusion Solar 17.27, HA reports 1.26 imported and Fusion Solar 1.33. The graphic on the right is the total for today which is also close. Here you can see the second dial reporting my home battery percentage correctly :)

So ... I'm getting closer and closer :-)
